### PR TITLE
Feat: re check payload history

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## What's this PR do ?

- [x] re-check payload history

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [ ] It was tested in local success?
- [ ] Fill link PR into ticket and the opposite
- [ ] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr></table>

File: app/Http/Controllers/PayloadHistoryController.php [hunk 1]
```diff
@@ -93,4 +93,61 @@ public function destroy(Request $request, PayloadHistory $history)
             ]);
         }
     }
+
+    public function recheck($id) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadHistoryController.php</br>**Line**: 97</br>Opening brace should be on a new line

File: app/Http/Controllers/PayloadHistoryController.php [hunk 2]
```diff
@@ -93,4 +93,61 @@ public function destroy(Request $request, PayloadHistory $history)
...
+        }
+        for ($i = 0; $i < count($result); $i++) {
+            try {
+                $this->getValues($params, $result[$i]);
+            } catch (\Throwable $th) {
+                array_push($errorFields, trim($result[$i]));
+                continue;
+            }
+        }
+        if(count($errorFields) != 0) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadHistoryController.php</br>**Line**: 117</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Http/Controllers/PayloadHistoryController.php [hunk 3]
```diff
@@ -93,4 +93,61 @@ public function destroy(Request $request, PayloadHistory $history)
...
+            return response()->json([
+                'error' => true,
+                'data' => array_unique($errorFields)
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadHistoryController.php</br>**Line**: 120</br>Each line in an array declaration must end in a comma

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadHistoryController.php [hunk 4]
```diff
@@ -93,4 +93,62 @@ public function destroy(Request $request, PayloadHistory $history)
             ]);
         }
     }
+
+    public function recheck($id)
+    {
+        $errorFields = [];
+        $result = [];
+        $payloadHistory = PayloadHistory::findOrFail($id);
+        $params = json_decode(json_encode(json_decode($payloadHistory->params)), true);
```
Comment (phucqd-1305) [inline app/Http/Controllers/PayloadHistoryController.php]: sao chỗ này encode vs decode nhiều thế

File: app/Http/Controllers/PayloadHistoryController.php [hunk 5]
```diff
@@ -93,4 +93,62 @@ public function destroy(Request $request, PayloadHistory $history)
...
+        $payloads = $payloadHistory->webhook->payloads;
+
+        for ($i = 0; $i < count($payloads); $i++) {
```
Comment (phucqd-1305) [inline app/Http/Controllers/PayloadHistoryController.php]: Sao k dùng foreach mà lại dùng for

Comment (nguyenthinhhn) [reply app/Http/Controllers/PayloadHistoryController.php]: em sửa r a.

File: app/Http/Controllers/PayloadHistoryController.php (hunk 4 above)
Comment (nguyenthinhhn) [reply app/Http/Controllers/PayloadHistoryController.php]: như thế mới ra đk dạng mình cần a

**Comment by framgiaciv3** [comment]
No data coverage!

**Comment by framgiaciv3** [comment]
No data coverage!

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadHistoryController.php (hunk 4 above)
Comment (phucqd-1305) [reply app/Http/Controllers/PayloadHistoryController.php]: Thế thì chỉ cần json_decode($payloadHistory->params, true); thôi chứ nhỉ. thử xem được không

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

File: routes/web.php [hunk 6]
```diff
@@ -40,6 +40,7 @@
     Route::put('webhooks/change_status', 'WebhookController@changeStatus');
     Route::resource('webhooks', 'WebhookController');
     Route::resource('history', 'PayloadHistoryController')->only(['show', 'index', 'destroy']);
+    Route::get('history/recheck/{id}', 'PayloadHistoryController@recheck')->name('history.recheck');
```
Comment (phucqd-1305) [inline routes/web.php]: Recheck dùng post đi em, với kiểm tra xem permission cho mỗi id chưa, tức là user chỉ được gọi api recheck cho cái id mà người ấy tạo ra thôi, trừ admin.

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**